docs/cody/model-configuration/index.mdx

```diff
@@ -0,0 +1,553 @@
+  }
+}
+```
+
+In the example above:
+
+-   Sourcegraph-supplied models are enabled (`sourcegraph` is not set to `null`).
+-   The `"openai"` provider configuration is overridden to be accessed using your own key, with the provider API accessed
+    directly via the specified `endpoint` URL. In contrast, models from the `"anthropic"` provider are accessed through the Cody Gateway.
+
+For additional examples, refer to the [examples page](/cody/model-configuration/examples).
+
+### Server-side Config
+
+The most important part of a provider's configuration is the `"serverSideConfig"` field, which defines how the LLM models
+should be invoked—that is, which external service or API will handle the LLM requests.
+
+Sourcegraph natively supports several types of LLM API providers. The current set of supported providers includes:
+
+| Provider type        | Description                                                                                                 |
+| -------------------- | ----------------------------------------------------------------------------------------------------------- |
+| `"sourcegraph"`      | [Cody Gateway](/cody/core-concepts/cody-gateway), offering access to various models from multiple services. |
+| `"openaicompatible"` | Any OpenAI-compatible API implementation.                                                                   |
+| `"awsBedrock"`       | [Amazon Bedrock](https://docs.aws.amazon.com/bedrock/)                                                      |
+| `"azureOpenAI"`      | [Microsoft Azure OpenAI](https://azure.microsoft.com/en-us/products/ai-services/openai-service/)            |
+| `"anthropic"`        | [Anthropic](https://www.anthropic.com)                                                                      |
+| `"fireworks"`        | [Fireworks AI](https://fireworks.ai)                                                                        |
+| `"google"`           | [Google Gemini](http://cloud.google.com/gemini) and [Vertex](https://cloud.google.com/vertex-ai/)           |
+| `"openai"`           | [OpenAI](http://platform.openai.com)                                                                        |
+| `"huggingface-tgi"`  | [Hugging Face Text Generation Interface](https://huggingface.co/docs/text-generation-inference/en/index)    |
+
+The configuration for `serverSideConfig` varies by provider type.
+
+For examples, refer to the [examples page](/cody/model-configuration/examples).
+
+## Model Overrides
+
+With a provider defined (either a Sourcegraph-supplied provider or a custom provider configured via the `providerOverrides`
+field), custom models can be specified for that provider by adding them to the `"modelConfiguration.modelOverrides"` section.
+
+This field is an array of items, each with the following fields:
+
+-   `modelRef` - Uniquely identifies the model within the provider namespace.
+    -   A string in the format `${providerId}::${apiVersionId}::${modelId}`.
```

Comment:
Might be worth calling out here (or somewhere) that the components (`providerId`, `apiVersionId`, and `modelId`) must not contain colon (`:`) characters an that our implementation always replaces colons with an underscore (`_`) when generating the `modelRef`.